src/main/java/pl/pjwstk/woloappapi/security/AuthenticationController.java

```diff
@@ -0,0 +1,88 @@
+package pl.pjwstk.woloappapi.security;
+
+import lombok.AllArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+import pl.pjwstk.woloappapi.model.UserEntity;
+import pl.pjwstk.woloappapi.repository.RoleRepository;
+import pl.pjwstk.woloappapi.repository.UserRepository;
+
+import javax.validation.Valid;
+import java.net.URI;
+
+@RestController
+@AllArgsConstructor
+@RequestMapping("auth")
+public class AuthenticationController {
+    @Autowired
+    private UserRepository userRepository;
+    @Autowired
+    private RoleRepository roleRepository;
+    @Autowired
+    private AuthenticationManager authenticationManager;
+    @Autowired
+    private TokenProvider tokenProvider;
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+    private static final Logger logger = LoggerFactory.getLogger(AuthenticationController.class);
+    @PostMapping("/login")
+    public ResponseEntity<?> authenticateUser(@Valid @RequestBody LoginRequest loginRequest) {
+        Authentication authentication = authenticationManager.authenticate(
+                new UsernamePasswordAuthenticationToken(
+                        loginRequest.getEmail(),
+                        loginRequest.getPassword()
+                )
+        );
+        SecurityContextHolder.getContext().setAuthentication(authentication);
+
+        String token = tokenProvider.createToken(authentication);
+        return ResponseEntity.ok(new AuthResponse(token));
+    }
+    @PostMapping("/signup")
+    public ResponseEntity<?> registerUser(@Valid @RequestBody SignUpRequest signUpRequest) {
+        if(userRepository.existsByEmail(signUpRequest.getEmail())) {
+            throw new BadRequestException("Email address already in use.");
+        }
+
+        UserEntity user = new UserEntity();
```

Comment:
jeśli nie mamy mappera, to w dalszej pracy warto pomyśleć o dodaniu, albo o użyciu Buildera tu.

```
  UserEntity user = UserEntity.Builder()
    .firstname(signUpRequest.getFirstname())
    .lastname(signUpRequest.getLastname())
    .phoneNumber(signUpRequest.getPhoneNumber())
    .role(roleRepository.findById(3L).orElse(null))
    .peselVerified(false)
    .adult(true)
    .email(signUpRequest.getEmail())
    .password(passwordEncoder.encode(signUpRequest.getPassword()))
    .build()
```